framework/src/modules/chain/bft/headers_list.js

```diff
@@ -126,6 +126,12 @@ class HeadersList {
 	}
 
 	get(height) {
+		if (
+			this.length === 0 ||
+			this._items[height - this.first.height] === undefined
+		) {
+			return undefined;
+		}
```

Comment:
That's the default behaviour of array. If index does not exists it will return `undefined`. That's why additional conditions not required here.

Reply:
if `this.first` is `undefined`, it will crash (which should not happen, but it happened on one of the node)

Reply:
Can we simply the condition in that case based with simple; 

```
if (this.length === 0) return undefined; 
```